aerc: 0.2.1 -> 0.3.0

d-xo opened the pull request:
###### Motivation for this change

version bump

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @tadeokondrak

tilpner commented:
This fails to start with

> Failed to load config: Can't find template - quoted_reply

jonringer reviewed (approved):
`nix-review` passes on NixOS
diff LGTM
commits LGTM
executable seems to work

```
[4 built, 7 copied (391.7 MiB), 99.2 MiB DL]
https://github.com/NixOS/nixpkgs/pull/73873
1 package were built:
aerc
```

jonringer commented:
oh, assumed there was a `aerc --config-path=../.` type option, didn't realize this was a regression :)

lovesegfault commented:
@xwvvvvwx Did you push the fix? is this ready for merging?

d-xo committed:
aerc: 0.2.1 -> 0.3.0

- bump version
- build with notmuch support
- substitute @SHAREDIR@ in aerc.conf for the `template-dirs` key
- place ncurses instead of ncurses.dev in PATH (cannot find infocmp otherwise)

xwvvvvwx commented:
Apologies for the delay. Rebased on master and made the following changes:

- build with notmuch support
- substitute @SHAREDIR@ in aerc.conf for the `template-dirs` key
 - place ncurses instead of ncurses.dev in PATH (the binary in `result` was otherwise unable to find `infocmp`)

jonringer reviewed (approved):
`nix-review` passes on NixOS
diff LGTM
commits LGTM

```
https://github.com/NixOS/nixpkgs/pull/73873
1 package were built:
aerc
```

jonringer commented:
@GrahamcOfBorg build aerc
